nevergrad/functions/multiobjective/core.py

```diff
@@ -45,17 +53,23 @@ def compute_aggregate_loss(self, losses: ArrayLike, *args: Any, **kwargs: Any) -
         """Given parameters and the multiobjective loss, this computes the hypervolume
         and update the state of the function with new points if it belongs to the pareto front
         """
-        # We compute the hypervolume
+        if self._auto_bound > 0:
+            self._upper_bounds = np.array(losses) if self._auto_bound == self._bound_budget else np.maximum(self._upper_bounds, np.array(losses))
```

Comment:
simpler to read:

In the `__init__`:
```python
self._auto_bound = 0
self._auto_upper_bounds = np.array([-float('inf')])  
self._auto_lower_bounds = np.array([float('inf')])
if upper_bounds is None:
     self._auto_bound = 100
```
(no need of `self._bound_budget` anymore)

then at the beginning of compute aggregate loss (to avoid casting it again and again):
```python
losses = np.array(losses, copy=False)
```
and finally
```python
self._auto_upper_bounds = np.maximum(self._auto_upper_bounds, losses)
self._auto_lower_bounds = np.minimum(self._auto_lower_bounds, losses)
```